File: backend/src/main/java/store/andefi/core/entity/Account.java
package store.andefi.core.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import store.andefi.core.enums.Role;

import java.time.Instant;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Account {
    private UUID id;
    private String email;
    private String password;
    private String fullName;
    private String phoneNumber;
    private Instant verifiedAt;
    private Role role;
    private ShippingAddress shippingAddress;
}

File: frontend/app/src/main/java/store/andefi/data/local/entity/Account.kt
package store.andefi.data.local.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "account")
data class Account(
    @PrimaryKey val id: String,
    @ColumnInfo("full_name") val fullName: String,
    val email: String,
    val username: String,
    @ColumnInfo("jwt_token") val jwtToken: String
)